android_splash_screen/android/app/src/main/kotlin/com/example/splash_screen_sample/MainActivity.kt

```diff
@@ -0,0 +1,202 @@
+
+  var flutterUIReady : Boolean = false
+  var initialAnimationFinished : Boolean = false
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+
+    super.onCreate(savedInstanceState)
+
+    // This activity will be handling the splash screen transition
+    val splashScreen = installSplashScreen()
+
+    // The splashscreen goes edge to edge, so for a smooth transition to our app, we also
+    // want to draw edge to edge.
+    WindowCompat.setDecorFitsSystemWindows(window, false)
+    val insetsController = WindowCompat.getInsetsController(window, window.decorView)
+    insetsController?.isAppearanceLightNavigationBars = true
+    insetsController?.isAppearanceLightStatusBars = true
+
+
+    // The content view needs to set before calling setOnExitAnimationListener
+    // to ensure that the SplashScreenView is attach to the right view root.
+    val rootLayout= findViewById(android.R.id.content) as FrameLayout
+    View.inflate(this, R.layout.main_activity_2, rootLayout)
+
+
+    ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.container)) { v, i ->
+      val insets = i.getInsets(WindowInsetsCompat.Type.systemBars())
+      v.setPadding(insets.left, insets.top, insets.right, insets.bottom)
+      i.inset(insets)
+    }
+
+    // Setting an OnExitAnimationListener on the splash screen indicates
+    // to the system that the application will handle the exit animation.
+    // The listener will be called once the app is ready.
+    splashScreen.setOnExitAnimationListener { splashScreenViewProvider ->
+      onSplashScreenExit(splashScreenViewProvider)
+    }
+
+  }
+
+  override fun onFlutterUiDisplayed(){
+    flutterUIReady = true
+
+    if (initialAnimationFinished) {
+      hideSplashScreenAnimation()
+    }
+  }
+
+  override fun onFlutterUiNoLongerDisplayed(){
+    flutterUIReady = false
+  }
+
+  /**
+   * Hides the splash screen only when the entire animation has finished and the Flutter UI is ready to display
+   */
+  private fun hideSplashScreenAnimation(){
+    val splashView = findViewById(R.id.container) as ConstraintLayout
+    splashView
+      .animate()
+      .alpha(0.0f)
+      .setDuration(SPLASHSCREEN_FINAL_ANIMATION_ALPHA_ANIMATION_DURATION)
+      .setListener(
+        object: Animator.AnimatorListener {
+          override fun onAnimationStart(animation: Animator) {}
+          override fun onAnimationEnd(animation: Animator) {}
+          override fun onAnimationCancel(animation: Animator) {}
+          override fun onAnimationRepeat(animation: Animator) {}
+        })
+  }
+
+  /**
+   * Handles the transition from the splash screen to the application
+   */
+  private fun onSplashScreenExit(splashScreenViewProvider: SplashScreenViewProvider) {
+    val accelerateInterpolator = FastOutLinearInInterpolator()
+    val splashScreenView = splashScreenViewProvider.view
+    val iconView = splashScreenViewProvider.iconView
+
+    // We'll change the alpha of the main view
```

Comment:
Styling nit: Avoid using first-person pronouns like "We" or "I" in comments or docs. An easy way to do this is to either leave the subject of the sentence out entirely ("Change the alpha of the main view.") or to use "the app" or "this code" as the subject instead.